src/Moq/Matchers/ParamArrayMatcher.cs

```diff
@@ -16,7 +16,7 @@ public ParamArrayMatcher(NewArrayExpression expression)
 			if (expression != null)
 			{
 				this.matchers = expression.Expressions
-					.Select(e => MatcherFactory.CreateMatcher(e)).ToArray();
+					.Select(e => MatcherFactory.CreateMatcher(e).Item1).ToArray();
```

Comment:
As above. We're losing `.Item2` here (i.e. the expression from which the matcher instance is derived).